### DISCUSSION TITLE
编辑器原生支持 mermaid，效果可参考飞书编辑器

### DISCUSSION THREAD

**Discussion by yuanzhixiang** [halo-dev/halo#4791] [Ideas]
### 你当前使用的版本

2.7.0

### 描述一下此特性

程序员博客中经常需要画各种图，而这些图如果以文字的形式进行保存会非常方便易于维护，mermaid 就是这样一款工具，建议在 halo 编辑器中提供支持。

下面是飞书编辑器中选择 mermaid 和使用的效果：

![image](https://github.com/halo-dev/halo/assets/25138766/f313216c-a7a5-415d-87fa-1a3e514b49e9)

![image](https://github.com/halo-dev/halo/assets/25138766/239ac77a-7f64-4408-a319-6384c81e0442)

**Comment by ruibaby** [comment]
我更推荐像 https://github.com/halo-sigs/plugin-katex/ 一样，通过插件来实现对 Mermaid 的集成。

/area plugin
/area editor
/kind feature

**Comment by yuanzhixiang** [comment]
> 我更推荐像 https://github.com/halo-sigs/plugin-katex/ 一样，通过插件来实现对 Mermaid 的集成。
> 
> /area plugin /area editor /kind feature

嗯，是的，这样会更灵活

**Comment by xinkeng0** [comment]
有兴趣，我会尝试

**Comment by xinkeng0** [comment]
~~抱歉迟到了。你可以试用pre版本，已支持mermaid编辑与展示。~~

~~https://github.com/gengxiaoxiaoxin/plugin-text-diagram~~

release：https://github.com/halo-sigs/plugin-text-diagram

**Reply by Binaryify** [reply]
为什么我的可以预览，但是文章页还是纯文本

**Reply by xinkeng0** [reply]
plantuml?这个目前还不能在文章页展示，因为需要额外部署一个PlantUML-Server

**Reply by xinkeng0** [reply]
pre版本有一些内部错误，正式版已发布

**Reply by ruibaby** [reply]
PlantUML 可以通过库的形式调用吗？这样的话应该就可以放在插件中。目前我看好像是用的 plantuml.com 的服务去拿到的图片。

**Reply by xinkeng0** [reply]
可以，但是有一些细节性问题，稍后我会在[plugin-text-diagram](https://github.com/halo-sigs/plugin-text-diagram)#issue中说明。

**Comment by xinkeng0** [comment]
> 可以，但是有一些细节性问题，稍后我会在[plugin-text-diagram](https://github.com/halo-sigs/plugin-text-diagram)#issue中说明。

see https://github.com/halo-sigs/plugin-text-diagram/issues/7
